File: xml/System/snippets/csharp/string.split/identify.cs
using System;

namespace Split
{
    class Identify
    {
        private static void SplitWithCharAndInt()
        {
            // <Snippet3>
            string phrase = "The quick  brown fox";

            _ = phrase.Split(default(Char[]), 3, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split((char[])null, 3, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split(null as char[], 3, StringSplitOptions.RemoveEmptyEntries);
            // </Snippet3>
        }

        private static void SplitWithStringAndInt()
        {
            // <Snippet4>
            string phrase = "The quick  brown fox";

            _ = phrase.Split(default(string[]), 3, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split((string[])null, 3, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split(null as string[], 3, StringSplitOptions.RemoveEmptyEntries);
            // </Snippet4>
        }

        private static void SplitWithChar()
        {
            // <Snippet5>
            string phrase = "The quick  brown fox";

            _ = phrase.Split(default(Char[]), StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split((char[])null, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split(null as char[], StringSplitOptions.RemoveEmptyEntries);
            // </Snippet5>
        }

        private static void SplitWithString()
        {
            // <Snippet6>
            string phrase = "The quick  brown fox";

            _ = phrase.Split(default(string[]), StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split((string[])null, StringSplitOptions.RemoveEmptyEntries);

            _ = phrase.Split(null as string[], StringSplitOptions.RemoveEmptyEntries);
            // </Snippet6>
        }
    }
}

File: snippets/fsharp/System/String/Split/identify.fs
module identify
open System

let splitWithCharAndInt () =
    // <Snippet3>
    let phrase = "The quick  brown fox"
    
    phrase.Split(Unchecked.defaultof<char[]>, 3, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split(null :> char[], 3, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split((null: char[]), 3, StringSplitOptions.RemoveEmptyEntries) |> ignore
    // </Snippet3>

let splitWithStringAndInt () =
    // <Snippet4>
    let phrase = "The quick  brown fox"

    phrase.Split(Unchecked.defaultof<string[]>, 3, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split(null :> string[], 3, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split((null: string[]), 3, StringSplitOptions.RemoveEmptyEntries) |> ignore
    // </Snippet4>

let splitWithChar () =
    // <Snippet5>
    let phrase = "The quick  brown fox"

    phrase.Split(Unchecked.defaultof<char[]>, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split(null :> char[], StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split((null: char[]), StringSplitOptions.RemoveEmptyEntries) |> ignore
    // </Snippet5>

let splitWithString () =
    // <Snippet6>
    let phrase = "The quick  brown fox"

    phrase.Split(Unchecked.defaultof<string[]>, StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split(null :> string[], StringSplitOptions.RemoveEmptyEntries) |> ignore

    phrase.Split((null: string[]), StringSplitOptions.RemoveEmptyEntries) |> ignore
    // </Snippet6>